src/base/stun.spec.ts

```diff
@@ -54,15 +94,23 @@ describe('test STUN', function () {
   })
 
   it('should perform a STUN request', async function () {
-    const multiAddrs = servers.map((server: ServerType) =>
-      Multiaddr.fromNodeAddress(nodeToMultiaddr(server.socket.address()), 'udp')
-    )
+    const multiAddrs = servers
+      .slice(1)
+      .map((server: ServerType) => Multiaddr.fromNodeAddress(nodeToMultiaddr(server.socket.address()), 'udp'))
 
     const result = await getExternalIp(multiAddrs, servers[0].socket)
 
+    // FIXME this fails when running behind a bidirectional NAT
     assert(result != undefined, `STUN request must be successful`)
 
-    assert(servers[0].socket.address().port === result.port, 'Ports should match')
+    const u8aAddress = ipToU8aAddress(result.address, 'IPv4')
+
+    // FIXME this fails when running behind a bidirectional NAT
+    assert(
```

Comment:
use multiple `assert`s for easier test outcome reading

Reply:
fixed